### ISSUE TITLE
[Feature] Ability to specify the securityContext for the MariaDB instance pod

### ISSUE THREAD

**Issue by Vidalee** [mmontes11/mariadb-operator#62]
**Is your feature request related to a problem? Please describe.**
I wanted to experiment with the operator using AWS EFS volumes, but the folders in this type of volume are owned by user 1000.
When MariaDB starts, it chown the /var/lib/mysql to its user. But it doesn't have the permission to do so, since the folders are owned by user 1000. (`chown: changing ownership of '/var/lib/mysql/': Operation not permitted`).

Running the pod with 
```yaml
securityContext:
  runAsUser: 1000
```
would solve this issue. Unfortunately, while it's possible to define the securityContext for the controller pod and the controller container in the `values.yaml` file, it doesn't seem like it's possible for the MariaDB pod.

**Describe the solution you'd like**
The possibility to define the securityContext of the MariaDB pod in the `values.yaml` file.

**Describe alternatives you've considered**
Exploring the template files of the Helm chart to hardcode the securityContext, without success.

**Additional context**
None.

**Environment details**:
- Kubernetes version: v1.25.6-eks-48e63af
- mariadb-operator version: v0.0.9
- Install method: helm
- Install flavour: minimal

**Comment by mmontes11** [comment]
Hey @Vidalee !

Totally agreed! We are missing this feature. We could support both the `podSecurityContext` (Pod level) and `securityContext` (container level):

```yaml

apiVersion: mariadb.mmontes.io/v1alpha1
kind: MariaDB
metadata:
  name: mariadb
spec:
...
  podSecurityContext:
     fsGroup: 1
     runAsNonRoot: true
  securityContext:
     allowPrivilegeEscalation: false
     capabilities:
       drop:
        - ALL
     readOnlyRootFilesystem: true
     runAsNonRoot: true
... 
``` 

This will be supported in the following release.

Thanks! 🙏🏻

**Comment by mmontes11** [comment]
Hey @Vidalee !

This has been implemented an will be released in the `v0.0.10` version:
- https://github.com/mmontes11/mariadb-operator/blob/main/config/samples/mariadb_v1alpha1_mariadb.yaml#L55..L59

However, I found myself struggling to not run as root using the default image ([oficial MariaDB image](https://hub.docker.com/_/mariadb/tags)), you may specify a different one, it's supported by the operator:
- https://github.com/mmontes11/mariadb-operator/blob/main/config/samples/mariadb_v1alpha1_mariadb.yaml#L16..L19

Thanks!

**Comment by Vidalee** [comment]
Thanks for the quick response, I tested the new feature and it works well ! 👍 

As a side not for anyone who might end up on this issue, you don't want to use EFS for MariaDB storage but EBS instead. EFS being a network file system it is too slow for MariaDB.
